### PR TITLE
Fix interactive prompt with --last

### PR DESCRIPTION
**Pull request by piyush-garg**
This will fix the issue of interactive prompt getting
started with --last flag in scenario of not using
resource or param flag. It should take the value from
last pipelinerun in case of no use of resource and
param flag. In case of using the flag, it will override
the value of last run but should never start
interactive prompt.

Change the resource selection question in interactive
prompt for better user experience.

Adds tests for the given scenario

Fixes https://github.com/tektoncd/cli/issues/297

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: pkg/cmd/pipeline/start_test.go [hunk 1]
```diff
@@ -356,6 +356,80 @@ func Test_start_pipeline_last(t *testing.T) {
...
+
+	cs := pipelinetest.Clients{
+		Pipeline: pClient,
+		Kube:     fakekubeclientset.NewSimpleClientset(),
+	}
+
+	p := &test.Params{Tekton: cs.Pipeline, Kube: cs.Kube}
+
+	pipeline := Command(p)
+	got, _ := test.ExecuteCommand(pipeline, "start", pipelineName,
```
Comment (sthaha) [inline pkg/cmd/pipeline/start_test.go]: are we testing the simplest case that  `-l` starts a new pipelinerun using the params, resources, sa from the latest `pipelinerun` ?

File: pkg/cmd/pipeline/start.go [hunk 2]
```diff
@@ -158,25 +158,27 @@ func (opt *startOptions) getInput(pname string) error {
 		return err
 	}
 
-	if len(opt.Resources) == 0 {
-		pres, err := getPipelineResources(cs.Tekton, opt.cliparams.Namespace())
-		if err != nil {
-			fmt.Fprintf(opt.stream.Err, "failed to list pipelineresources from %s namespace \n", opt.cliparams.Namespace())
-			return err
-		}
+	if opt.Last != true {
```
Comment (sthaha) [inline pkg/cmd/pipeline/start.go]: `!= true` => ` == false`

Comment (sthaha) [reply pkg/cmd/pipeline/start.go]: I think this creates a bit too much nesting. can this be simplified?

File: pkg/cmd/pipeline/start_test.go (hunk 1 above)
Comment (piyush-garg) [reply pkg/cmd/pipeline/start_test.go]: yes, I have added test for this basic scenario now. Resource and Param from the previous run.

File: pkg/cmd/pipeline/start.go (hunk 2 above)
Comment (piyush-garg) [reply pkg/cmd/pipeline/start.go]: Improved a bit

**Comment by piyush-garg** [comment]
Go.mod is coming from master

File: pkg/cmd/pipeline/start_test.go [hunk 3]
```diff
@@ -356,6 +356,77 @@ func Test_start_pipeline_last(t *testing.T) {
...
+		"--last",
+		"-nns")
```
Comment (sthaha) [inline pkg/cmd/pipeline/start_test.go]: nit: -n ns

File: pkg/cmd/pipeline/start_test.go [hunk 4]
```diff
@@ -356,6 +356,77 @@ func Test_start_pipeline_last(t *testing.T) {
...
+	expected := "Pipelinerun started: random\n\nIn order to track the pipelinerun progress run:\ntkn pipelinerun logs random -f\n"
+	test.AssertOutput(t, expected, got)
+
+	pr, err := cs.Pipeline.TektonV1alpha1().PipelineRuns(p.Namespace()).Get("random", v1.GetOptions{})
+
+	if err != nil {
+		t.Errorf("Error getting pipelineruns %s", err.Error())
+	}
+
+	for _, v := range pr.Spec.Resources {
```
Comment (sthaha) [inline pkg/cmd/pipeline/start_test.go]: nit: pretty sure there will be a better way to obtain the resource `"git-repo"` but that can be a different patch. 
@vdemeester

File: pkg/cmd/pipeline/start_test.go (hunk 3 above)
Comment (piyush-garg) [reply pkg/cmd/pipeline/start_test.go]: resolved this.

**Comment by piyush-garg** [comment]
@sthaha These `go.mod` changes are coming from the latest master. My branch is up to date with the master.
